lib/cluster.js

```diff
@@ -27,7 +28,19 @@ function Worker(options) {
   if (options === null || typeof options !== 'object')
     options = {};
 
-  this.suicide = undefined;
+  this.voluntaryExit = undefined;
+
+  Object.defineProperty(this, 'suicide', {
+    get: function() {
+      // TODO: Print deprecation message.
+      return this.voluntaryExit;
+    },
+    set: function(val) {
+      // TODO: Print deprecation message.
```

Comment:
Should be slated for a major version. Honestly, we should just merge this and then open another PR right away with the deprecation warning since `master` is v6.